### PR TITLE
trigger deployment when a tag with "release-" prefix has been added.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: this PR adds functionality that will trigger a deployment when a tag with "release-" prefix has been added.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -84,20 +84,34 @@ workflows:
...
           requires:
             - install-dependencies
+          filters:
+            tags:
+              only: /release.*/
       - build:
           requires:
             - install-dependencies
+          filters:
+            tags:
```
Comment (p0lyn0mial) [inline .circleci/config.yml]: unfortunately we have to add `tags` to all jobs, as stated in the documentation:
> a job must have a filters tags section to run as a part of a tag push and all its transitively dependent jobs must also have a filters tags section

see also:
https://circleci.com/docs/2.0/workflows/#git-tag-job-execution

Comment (mrIncompetent) [reply .circleci/config.yml]: Not pretty, but i'm fine with doing so

File: .circleci/config.yml [hunk 2]
```diff
@@ -84,20 +84,34 @@ workflows:
...
+              only: /release.*/
       - publish:
           requires:
             - build
           filters:
             branches:
               only:
                 - master
+            tags:
+              only: /release.*/
```
Comment (p0lyn0mial) [inline .circleci/config.yml]: unfortunately CircleCI treats `tags` and `branches` separately. I'm afraid we cannot say IF branch=master AND tag=release.* then act. Which means that if we have a match on a tag then we will fire off `publish` job.

an example:
https://circleci.com/workflow-run/a3bd9483-36f8-42a2-8fd8-c97a51b92cf7

Comment (p0lyn0mial) [reply .circleci/config.yml]: Which means that `publish` can be triggered on any branch not just master.

Comment (mrIncompetent) [reply .circleci/config.yml]: So this code means, we only publish when the tag begins with `release.` ?

Comment (p0lyn0mial) [reply .circleci/config.yml]: yes, but also on branch=master push (as before)

Comment (mrIncompetent) [reply .circleci/config.yml]: The prefix should only be `v` as we'll only tag with the version.
For example:
- `v0.2.1`
- `v0.3.0`
etc.

Comment (p0lyn0mial) [reply .circleci/config.yml]: we could however read a branch name directly in `publish` job and fail if there is no match.

**Comment by p0lyn0mial** [comment]
FYI: tag will be derived from `CIRCLE_TAG` environment.

**Pull request merged**